utilities/rpp-unittests/HIP/BatchPD_hip.cpp

```diff
@@ -0,0 +1,749 @@
+	for(i = 0 ; i < images ; i++)
+	{
+		gamma[i] = 0.8;
+	}
+	Rpp32f fogValue[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		fogValue[i] = 0.7;
+	}
+	Rpp32u flipAxis[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		flipAxis[i] = 2;
+
+	}
+	Rpp32f exposureFactor[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		exposureFactor[i] = 3;
+	}
+	Rpp32s adjustmentValue[images];
+        Rpp32u extractChannelNumber[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		adjustmentValue[i] = 70;
+                extractChannelNumber[i] = 1;
+	}
+        Rpp32f alpha[images];
+	Rpp32f beta[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		alpha[i] = 1;
+		beta[i] = 0;
+	}
+	Rpp32f angle[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		angle[i] = 50;
+	}
+
+    Rpp32f affine_array[6*images];
+    for(i = 0; i < images; i=i+6)
+    {
+        affine_array[i] = 1.0;
+        affine_array[i+1] = 1.5;
+        affine_array[i+2] = 2.0;
+        affine_array[i+3] = 2.0;
+        affine_array[i+4] = 2.5;
+        affine_array[i+5] = 3.0;
+    }
+    
+    Rpp32u x1[images];
+    Rpp32u x2[images];
+    Rpp32u y1[images];
+    Rpp32u y2[images];
+
+    for(i = 0; i < images; i++)
+    {
+       x1[i] = 100;
+       x2[i] = 300;
+       y1[i] = 200;
+       y2[i] = 350;
+    }
+       Rpp32u xRoiBegin[images];
+	Rpp32u xRoiEnd[images];
+	Rpp32u yRoiBegin[images];
+	Rpp32u yRoiEnd[images];
+        Rpp32u mirrorFlag[images];
+
+	for(i = 0 ; i < noOfImages ; i++)
+	{
+		xRoiBegin[i] = 50;
+		yRoiBegin[i] = 50;
+        	xRoiEnd[i]  = 200;
+        	yRoiEnd[i]  = 200;
+        	mirrorFlag[i] = 0;
+        //dstSize[i].height = yRoiEnd[i];
+        //dstSize[i].width =  xRoiEnd[i];
+	}
+        Rpp32u crop_pos_x[images];
+	Rpp32u crop_pos_y[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		crop_pos_x[i] = 100;
+		crop_pos_y[i] = 100;
+	}
+	Rpp32f hueShift[images];
+	Rpp32f saturationFactor[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		
+		hueShift[i] = 10;
+		saturationFactor[i] = 5;
+	}
+        Rpp8u minThreshold[images];
+	Rpp8u maxThreshold[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		minThreshold[i] = 10;
+		maxThreshold[i] = 30;
+	}
+        Rpp32u numOfPixels[images];
+        Rpp32u gaussianKernelSize[images];
+	Rpp32f kValue[images];
+	Rpp8u threshold[images];
+        Rpp32f threshold1[images];
+	Rpp32u nonmaxKernelSize[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		numOfPixels[i] = 4;
+		gaussianKernelSize[i] = 7;
+		kValue[i] = 1;
+		threshold[i] = 15;
+            	threshold1[i] = 10;
+		nonmaxKernelSize[i] = 5;
+	}
+        Rpp32u sobelType[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		sobelType[i] = 1;
+	}
+        Rpp8u min[images];
+	Rpp8u max[images];
+	for(i = 0 ; i < images ; i++)
+	{
+		min[i] = 10;
+		max[i] = 30;
+	}
+        const int size_perspective = images * 9;
+    Rpp32f perspective[64 * 9];
+    for(i = 0 ; i < images ; i++)
+	{
+        perspective[0 + i * 9] = 1;
+        perspective[1 + i * 9] = 0.5;
+        perspective[2 + i * 9] = 0;
```

Comment:
Codacy found an issue: [Array 'perspective[576]' accessed at index 893, which is out of bounds.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)